Wikipedia/UI-V5/WMFSearchViewController.m

```diff
@@ -75,17 +75,32 @@ - (void)updateSearchStateAndNotifyDelegate:(WMFSearchState)state {
     }
 
     self.state = state;
-    [self updateRecentSearchesVisibility];
+    [self updateRecentSearchesVisibility:YES];
 
     [self.delegate searchController:self searchStateDidChange:self.state];
 }
 
 - (void)updateRecentSearchesVisibility {
-    if ([self.searchBar.text length] == 0 && self.recentSearchesViewController.recentSearchesItemCount > 0) {
-        [self.recentSearchesContainerView setHidden:NO];
-    } else {
-        [self.recentSearchesContainerView setHidden:YES];
+    [self updateRecentSearchesVisibility:YES];
+}
+
+- (void)updateRecentSearchesVisibility:(BOOL)animated {
+    BOOL hideRecentSearches =
+        [self.searchBar.text length] > 0 || self.recentSearchesViewController.recentSearchesItemCount == 0;
+
+    /*
+       HAX: Need to show/hide superviews since recent & results are in the same container. should use UIViewController
+          containment/transition API instead in the future.
+     */
+    if ((self.recentSearchesViewController.view.superview.alpha == 0) == hideRecentSearches) {
```

Comment:
addressed in f6d8b8f.

> 1. The line it self pretty hard to read. We should favor readability over compactness. two equality comparisons makes this hard to grok.

True, refactored the visibility check into a getter method.

> 2. At a minimum we can encapsulate this logic in a well named method to make it obvious what/why this is being compared.

Yep :point_up: 

> 3. Isn't self.recentSearchesViewController.view.superview just the container view?

Yep, didn't realize there were outlets for it.

> 4. If the answer is yes, then couldn't we can create an IBOutlet to hold the relevant view to make the intent more obvious?

Already existed :point_up:

> 5. Should we be capturing this state in an ivar (a `BOOL`) to be more explicit instead of relying on the alpha state which may be fragile?

turns out there are some issues w/ this.. looking into it now